### DISCUSSION TITLE
Can we clear force?

### DISCUSSION THREAD

**Discussion by keiohta** [haosulab/SAPIEN#69] [Q&A] [answered]
Hi author, I encountered weird movements when I try to control an articulated object as in the attached movie. In the movie, I want to change only one joint (`joint_2`), but both joints move unexpectedly. The object is loaded from the Part-Net Mobility dataset whose id is `10068`. I think the reason why both links move is that the quality of collision meshes provided by the dataset is not so good, and it seems there happens an unexpected collision between two links. So, we can't fundamentally solve it unless we modify the mesh files. Instead, I think resetting forces that work for each link can be a good alternative (stop unexpected moves by clearing forces). So, could you tell me how to clear forces?

https://user-images.githubusercontent.com/7234712/157371979-68cee34c-5c54-48f9-a073-0312c52bad90.mp4

**Comment by keiohta** [comment]
Ah, I might misunderstand the implementation of SAPIEN. We might clear velocities instead of forces (I guess forces are applied only when a collision happens and will be cleared at the next step)? Anyway, I would like to ask your opinion on how to deal with this situation. Thank you.

**Comment by fbxiang** [comment]
Collision in SAPIEN should just work without any additional code.
From my experience, fridges in the dataset are modeled really bad. Maybe you can click on the show on the bottom right to investigate the collision shapes of this fridge.

**Reply by keiohta** [reply]
Thanks for the quick reply!
As you say, the collision shape for the right door is not as expected... To deal with these situations, can we ignore the collision between these two links without modifying URDFs? More generally, can we ignore collisions between all movable links?

![image](https://user-images.githubusercontent.com/7234712/157375351-251f7a75-e82e-4087-b1af-428b13a22812.png)

**Comment by fbxiang** [comment, marked as answer]
Yes. Here is an example of ignoring all collisions between movable links.
https://github.com/haosulab/ManiSkill/blob/af155e57bacfa43626be59e87e5b0439677198cc/mani_skill/env/open_cabinet_door_drawer.py#L181-L189

**Reply by keiohta** [reply]
I just tried the solution and confirmed the links don't collide with each other. Thank you!!
